iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2460,7 +2459,7 @@ static void ProcessPendingTelemetryMessages(PMQTTTRANSPORT_HANDLE_DATA transport
 
         if (((current_ms - msg_detail_entry->msgPublishTime) / 1000) > RESEND_TIMEOUT_VALUE_MIN)
         {
-            if (msg_detail_entry->retryCount >= MAX_SEND_RECOUNT_LIMIT)
+            if (((current_ms - msg_detail_entry->msgCreationTime) / 1000) >= MSG_TIMEOUT_VALUE_MIN)
```

Comment:
I would kind of lean towards the simpler "this message has 2 minutes to get PUBACK'd or else", esp. since we should be refining our documentation with all these changes at some point [here](https://github.com/Azure/azure-iot-sdk-c/blob/main/doc/connection_and_messaging_reliability.md), but I don't have that strong opinions here.